compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1508,6 +1508,20 @@ private void createDummyTypeDefSymbol(BLangTypeDefinition typeDef, SymbolEnv env
         // This is only to keep the flow running so that at the end there will be proper semantic errors
         typeDef.symbol = Symbols.createTypeSymbol(SymTag.TYPE_DEF, Flags.asMask(typeDef.flagSet),
                 names.fromIdNode(typeDef.name), env.enclPkg.symbol.pkgID, typeDef.typeNode.type, env.scope.owner);
+        typeDef.symbol.scope = env.scope;
+
+        // Todo - Add more kinds
+        switch (typeDef.typeNode.type.tag) {
+            case TypeTags.RECORD:
+                typeDef.symbol.kind = ((BLangRecordTypeNode) typeDef.typeNode).symbol.kind;
+                ((BLangRecordTypeNode) typeDef.typeNode).symbol.scope = env.scope;
+                break;
+            case TypeTags.OBJECT:
+                typeDef.symbol.kind = ((BLangObjectTypeNode) typeDef.typeNode).symbol.kind;
+                ((BLangObjectTypeNode) typeDef.typeNode).symbol.scope = env.scope;
+                break;
```

Comment:
We can merge records and objects into one by casting the typeNode to `BStructureTypeNode`

Reply:
Resolved with https://github.com/ballerina-platform/ballerina-lang/pull/12628/commits/0b0d27d10600a5f66c9bcc7fbff4c17a740a5c1b.